packages/stats/gbstats/gbstats.py

```diff
@@ -562,6 +598,46 @@ def process_single_metric(
     )
 
 
+def get_bandit_weights(
+    rows: pd.DataFrame,
+    var_id_map: VarIdMap,
+    metric: MetricSettingsForStatsEngine,
+    analysis: AnalysisSettingsForStatsEngine,
+) -> pd.DataFrame:
+    reduced = preprocess_analysis(rows, var_id_map, metric, analysis)
+    num_variations = reduced.at[0, "variations"]
+    bandit_prior = GaussianPrior(mean=0, variance=float(1e4), proper=True)
+    bandit_config = BanditConfig(prior_distribution=bandit_prior)
+
+    def get_bandit_weights_single(s: pd.Series) -> pd.Series:
+        s_final = pd.Series({"dimension": s["dimension"]})
+        # initialize weights as constant; then just return s if any update checks below fail
+        s_final["update_message"] = "did not update"
+        s_final["weights"] = np.full((num_variations,), 1 / num_variations).tolist()
+        # need proportion, samplemean, or ratio;
+        bandit_types = Union[ProportionStatistic, SampleMeanStatistic, RatioStatistic]
+        s0 = variation_statistic_from_metric_row(
+            row=s, prefix="baseline", metric=metric
+        )
+        if isinstance(s0, bandit_types):
```

Comment:
Feels like we should set the update/error message to be more informative if this fails.

Reply:
changed to "test statistic is not of type BanditStatistic"